addons/point_of_sale/models/pos_payment.py

```diff
@@ -75,7 +75,7 @@ def _create_payment_moves(self):
             journal = pos_session.config_id.journal_id
             payment_move = self.env['account.move'].with_context(default_journal_id=journal.id).create({
                 'journal_id': journal.id,
-                'date': fields.Date.context_today(payment),
+                'date': fields.Date.context_today(payment, order.date_order),
```

Comment:
Why is context_today required here?
Is there any use case where we may want to apply timezones on date_order? shouldn't it always be the same as the date_order no matter what?

Reply:
No clue, that's how it was done.

I think the reason is that you want to use the date of the place where the PoS session was and not the date of the server itself.

Reply:
I had a look, I think it doesn't change anything because we retrieve the order from the payment, which I think it means the context is propagated to the order too.